File: cpp/tool/gen_token.cpp
#include "tls_token.h"
#include <iostream>
#include <stdlib.h>

int main(int argc, char *argv[]) {
    if (argc != 6) {
        std::cout << "usage:\n" << argv[0] << "  appid  uid  expire  sk  sv_ver\n";
        return -1;
    }

    string appid = argv[1];
    string uid = argv[2];
    int expire = atoi(argv[3]);
    string sk = argv[4];
    string sk_ver = argv[5];

    string token, errmsg;
    int ret = GenToken(expire, appid, uid, sk, sk_ver, token, errmsg);
    if (ret != 0) {
        std::cout << "gen token failed, " << ret << ", " << errmsg << std::endl;
    } else {
        std::cout << token << std::endl;
    }
    return 0;
}

File: go/tool/gen_token.go
package main

import (
	"flag"
	"log"

	"github.com/xverse-lab/tls-sig-api/go/tls_sig_api"
)

func main() {
	var appid, uid, sk, skVer string
	var expire uint
	flag.StringVar(&appid, "appid", "", "")
	flag.StringVar(&uid, "uid", "", "")
	flag.StringVar(&sk, "sk", "", "secret key")
	flag.StringVar(&skVer, "skVer", "", "secret key version")
	flag.UintVar(&expire, "expire", 0, "token expire time")
	flag.Parse()

	if appid == "" || uid == "" || sk == "" || skVer == "" || expire == 0 {
		log.Fatalln("please input appid, uid, sk, skver, expire, you can input --help to get usage")
	}

	token, err := tls_sig_api.GenToken(appid, uid, sk, skVer, uint32(expire))
	if err != nil {
		log.Fatalf("gen token failed, %v", err)
	}
	log.Println(token)
}
